pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -994,6 +1015,22 @@ func (b BlockVolumeHandler) ReconstructVolumeHandler(_ volume.VolumePlugin, mapp
 	return volumeSpec, nil
 }
 
+// CheckVolumeExistence checks mount path directory if volume still exists, return true if volume is there
+// This method is handler for block volume
+func (b BlockVolumeHandler) CheckVolumeExistence(mountPath, volumeName string, mounter mount.Interface, uniqueVolumeName v1.UniqueVolumeName, podName volumetypes.UniquePodName, podUID types.UID, _ volume.AttachableVolumePlugin) (bool, error) {
+	blkutil := util.NewBlockVolumePathHandler()
+	if islinkExist, checkErr := blkutil.IsSymlinkExist(mountPath); checkErr != nil {
```

Comment:
Can you add a comment explaining why the block volumes need to check for both attachable and non-attachable? (compared to filesystem, which only checks for attachable)

Reply:
@msau42 
At the filesystem volume case, I believe that attachable check exists to avoid checking plugins like configmap who has directory under volumes but the directory isn't mounted.
```
e17c37b6-ce1b-11e7-aaf5-525400645a80/volumes/kubernetes.io~configmap
```

On the other hands, at the block volume case, either attachable or non-attachable, all of them should have pod device map path directory and symbolic link under the directory.
Therefore we don't need to check if attachable or not for block volume case.
```
./0e94706e-ca7b-11e7-b87a-525400645a80/volumeDevices/kubernetes.io~fc/block-pv001
```